geomstats/geometry/embedded_manifold.py

```diff
@@ -70,7 +128,102 @@ def projection(self, point):
 
         Returns
         -------
+        projected : array-like, shape=[..., dim_embedding]
             Projected point.
         """
-        raise NotImplementedError(
-            'projection is not implemented.')
+        pass
+
+
+class OpenSet(Manifold, ABC):
+    """Class for manifolds that are open sets of a vector space.
+
+    In this case, tangent vectors are identified with vectors of the ambient
+    space.
+
+    Parameters
+    ----------
+    dim: int
+        Dimension of the manifold. It is often the same as the ambient space
+        dimension but may differ in some cases.
+    ambient_manifold: Manifold
+        Ambient manifold that contains the manifold.
+    """
+
+    def __init__(self, dim, ambient_manifold, **kwargs):
+        super().__init__(dim, **kwargs)
+        self.ambient_manifold = ambient_manifold
+
+    def is_tangent(self, vector, base_point, atol=gs.atol):
+        """Check whether the vector is tangent at base_point.
+
+        Parameters
+        ----------
+        vector : array-like, shape=[..., dim]
+            Vector.
+        base_point : array-like, shape=[..., dim]
+            Point on the manifold.
+        atol : float
+            Absolute tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        is_tangent : bool
+            Boolean denoting if vector is a tangent vector at the base point.
+        """
+        return self.ambient_manifold.belongs(vector, atol)
+
+    def to_tangent(self, vector, base_point):
+        """Project a vector to a tangent space of the manifold.
+
+        Parameters
+        ----------
+        vector : array-like, shape=[..., dim]
+            Vector.
+        base_point : array-like, shape=[..., dim]
+            Point on the manifold.
+
+        Returns
+        -------
+        tangent_vec : array-like, shape=[..., dim]
+            Tangent vector at base point.
+        """
+        return self.ambient_manifold.projection(vector)
```

Comment:
Can the vector space have a projection that is different from the identity function? If not, maybe we should replace this with `return vector` to save some calls?

Reply:
I think I did this for SPD matrices, where `ambient_manifold=SymmetricMatrices` whose projection is not exactly identity

Reply:
👍